src/bundle.ts

```diff
@@ -0,0 +1,32 @@
+export default class Bundle {
+	public tarStream: NodeJS.ReadableStream
+
+	/**
+	 * deviceType: The slug of the device type that this bundle has been created
+	 * for
+	 */
+	public deviceType: string
+
+	/**
+	 * architecture: The architecture that this resin bundle is targeting
+	 */
+	public architecture: string
```

Comment:
The thing is we may not statically know all the supported device types soon as they will be coming from multiple dynamic sources.